x/mongo/driver/topology/server_options.go

```diff
@@ -58,6 +64,14 @@ func newServerConfig(opts ...ServerOption) *serverConfig {
 	return cfg
 }
 
+func (cfg *serverConfig) validate() error {
+	const tmpl = "options should be: %s; got: %s"
+	if cfg.maxConns != 0 && cfg.minConns > cfg.maxConns {
+		return fmt.Errorf(tmpl, "MaxConnections >= MinConnections", fmt.Sprintf("%d MinConnections, %d MaxConnections", cfg.minConns, cfg.maxConns))
```

Comment:
The validation error message should reference the user-settable config value name. For example:
> minPoolSize must be less than or equal to maxPoolSize, got minPoolSize=%d maxPoolSize=%d

Reply:
It's what I hesitate in. The setting function is named by `WithMinConnections` and `WithMaxConnecting`.

Reply:
Yeah, it is weird to reference a different field name than the one you're actually validating. Per our previous conversation, it might be less weird logic to move the default `maxConns` out of `newServerConfig` and into the client configuration options. That way you can do the validation and error messaging from the `ClientOptions` validator rather than a new server config validator.